Sources/SpeziLLMLocalDownload/LLMLocalDownloadManager.swift

```diff
@@ -80,24 +80,26 @@ public final class LLMLocalDownloadManager: NSObject {
     
     /// Starts a `URLSessionDownloadTask` to download the specified model.
     public func startDownload() {
```

Comment:
Maybe we should make this function `async` at some point? Then we could use the SpeziViews `AsyncButton` to trigger the start of the download and prevent all the unnecessary `Task { @MainActor in ... }` scheduling.

Reply:
But don't we still need `Task { @MainActor in ... }` whenever we want to change `state` since it is `@MainActor`. So the only task that we could get rid of would be the download task, but than we might not be able to cancel the task.

Reply:
Well, if the function is `async`, you can do `await MainActor.run {}`, meaning you actually await the execution of the closure, not just schedule one via `Task { @MainActor in ... }` (no proper cancellation, no guarantee of execution etc.)
We don't have to do that, just thought that this might be a nice addition.
Other option is to make the entire `startDownload()` function `@MainActor` (no `async` needed), but I'm not sure if that's desirable.
I'm also totally fine with leaving the code as is!

Reply:
@philippzagar - Interesting point here. I was imagining something like this, and it's kind of neat though 😄

```swift
func startDownload() async {
    if modelExists {
        await MainActor.run { state = .downloaded }
        return
    }
        
    cancelDownload()

    downloadTask = Task(priority: .userInitiated) {
        do {
            try await downloadModel()
            await MainActor.run { state = .downloaded }
        } catch {
            await MainActor.run {
                state = .error(
                    AnyLocalizedError(
                        error: error,
                        defaultErrorDescription: LocalizedStringResource(
                                "LLM_DOWNLOAD_FAILED_ERROR",
                                bundle: .atURL(from: .module)
                            )
                        )
                    )
                }
            }
        }

        await downloadTask?.value
    }
    
    @MainActor
    private func downloadModel() async throws {
        let repo = Hub.Repo(id: model.hubID)
        let modelFiles = ["*.safetensors", "config.json"]
        
        try await HubApi.shared.snapshot(from: repo, matching: modelFiles) { progress in
            self.state = .downloading(progress: progress)
        }
    }
```

Reply:
On `cancelDownload` we should update the state to `idle`, right?

Reply:
Thats an awesome solution thanks for providing this example! Its implemented now! 🚀